component.js

```diff
@@ -255,9 +259,21 @@ function factory (options) {
     }
 
     if (!hasShouldComponentUpdate(mixins)) {
-      mixins = [{
+      mixins.unshift({
         shouldComponentUpdate: _shouldComponentUpdate
-      }].concat(mixins);
+      });
+    }
+
+    if (!hasComponentWillReceiveProps(mixins)) {
```

Comment:
I'm thinking this one and line 237 should be added none the less, so I can provide my own `componentWillReceiveProps` and `componentWillUpdate` and still preserve this functionality?